modules/assim.batch/R/pda.utils.R

```diff
@@ -256,12 +256,23 @@ pda.load.priors <- function(settings, con, path.flag = TRUE) {
     for (i in seq_along(settings$pfts)) {
       
       files <- dbfile.check("Posterior", settings$pfts[[i]]$posteriorid, con, settings$host$name)
+      
       pid <- grep("post.distns.*Rdata", files$file_name)  ## is there a posterior file?
       if (length(pid) == 0) {
         pid <- grep("prior.distns.Rdata", files$file_name)  ## is there a prior file?
       }
       if (length(pid) > 0) {
         prior.paths[[i]] <- file.path(files$file_path[pid], files$file_name[pid])
+      } else {
```

Comment:
I respectfully disagree. I can see how 'files' would miss posteriors if they were on a different settings$host$name, but what I said on Gitter is that settings$pfts[[i]]$posteriorid is NULL in my case, and therefore files will be NULL.  More important, the code above had no check in place to deal with the case where files == NULL. It would cause pid to be NULL and the length(pid)==0 fix would stay NULL because files == NULL. The length(pid) > 0 check would be false, and thus prior.paths[[i]] would be NULL and load(prior.paths[[i]]) would attempt to load NULL.  My fix may be a hack, but the current code does need to catch this case and deal with it.  Also, my hack is the exact same logic that's used in get.parameter.samples to sample from the posteriors for ensemble runs (hence my argument that we need a shared set of functions for resolving posterior.ids throughout the code rather than different hacks in different places).